scripts/start.py

```diff
@@ -196,25 +196,33 @@ def main(args: Optional[Sequence[str]] = None) -> None:
             )
             initializer.populate_debug_data()
 
-        managed_web_browser = (
-            None if parsed_args.no_browser else
-            servers.create_managed_web_browser(PORT_NUMBER_FOR_GAE_SERVER))
-
-        if managed_web_browser is None:
+        try:
+            managed_web_browser = (
+                None if parsed_args.no_browser else
+                servers.create_managed_web_browser(PORT_NUMBER_FOR_GAE_SERVER))
+
+            if managed_web_browser is None:
+                common.print_each_string_after_two_new_lines([
+                    'INFORMATION',
+                    'Local development server is ready! You can access it by '
+                    'navigating to http://localhost:%s/ in a web '
+                    'browser.' % PORT_NUMBER_FOR_GAE_SERVER,
+                ])
+            else:
+                common.print_each_string_after_two_new_lines([
+                    'INFORMATION',
+                    'Local development server is ready! Opening a default web '
+                    'browser window pointing to it: '
+                    'http://localhost:%s/' % PORT_NUMBER_FOR_GAE_SERVER,
+                ])
+                stack.enter_context(managed_web_browser)
+        except Exception:
```

Comment:
This shadows the error. In some cases we might not care but it seems reasonable to at least print it out (perhaps as info, e.g. "Could not automatically start server because ...") in case the behaviour is not expected?

Reply:
I initially added that code but later removed thinking that we are logging the error in servers.py (using logging.exception). Shall I also add a print (common.print_each_string_after_two_new_lines) here?

Reply:
Actually, reading this more carefully -- I think that you shouldn't have the try/except at all in the create_managed_web_browser method(). Have the method raise an exception when it needs to and keep the try/except as small as possible (that's good practice). Then catch the exception here and do any logging needed.

Similarly I feel like this code can be simplified if you're raising an exception rather than returning None for the case where a browser can't be launched. I.e. if your function above either raises an exception or launches a browser then you don't need the None case any more, right?

Reply:
Added the code to print an ERROR message when exception occurs.

Reply:
I initially coded the way you are suggesting. I will update the code.

The 'None' is returned for operating systems other than Linux and Mac. If we remove that code (else block), then it will not raise an exception (as it will not try to open the browser) and will not print the message to instruct user to open a web browser to access the application. I think it will not be good to raise exception when OS is neither Linux or Mac. What do you say?

Reply:
I think it depends on how we are defining what "Exception" means. My understanding is that we are defining Exception here as "could not open web browser", so if we're going with that then we should raise an exception when we cannot open the web browser. (Probably on non-Linux/Mac systems the issue is just that we don't know what the command to automatically open the web browser is.)

An alternative design would result in more than those two states which I think is unnecessary complexity. What is your rationale for not raising an exception in the non-Linux/Mac case?

Reply:
I have added code to raise an exception with message 'Unable to identify the Operating System and therefore, unable to launch the web browser.' in servers.py when OS is non-Linux/Mac. 

The 'None' condition is required in start.py to handle the no_browser CLI argument. PTAL.